README.md

```diff
@@ -125,23 +128,65 @@ A single error is modeled to wrap the most common arguments of an error. Develop
 this model.
 
 ```kotlin
-class Error(
+@Serializable
+@SerialName("error")
+class DefaultError(
     val code: String,
     val title: String,
     val detail: String,
     val status: String
 )
 ```
 
-When using Retrofit, in the event of a network error a `HttpException` will be thrown. To extract the `Error` model from a response, you can
-use the `HttpException.asJsonXHttpException()` extension, which will return a `JsonXHttpException`, containing the original `response` as
-well as `errors` list.
+When using Retrofit, in the event of a network error a `HttpException` will be thrown. To extract the `DefaultError` model from a response, you can
+use the `HttpException.asJsonXHttpException<DefaultError>()` extension. The extension takes in a generic `DefaultError` which will return a `JsonXHttpException`, containing the original `response` as well as `errors` list.
+
+```kotlin
+try {
+    val person = io { sampleApiService.fetchPerson() }
+} catch (exception: HttpException) {
+    val errors = exception.asJsonXHttpException<DefaultError>().errors
+    errors?.first()?.let {
+        if (it is DefaultError) {
+            showError(it.code)
+        } else {
+            showError("Not a default error.")
+        } 
+    } ?: showError("Unable to parse to JsonXHttpException")
```

Comment:
This code example is too specific to our sample and doesn't show the full potential of the feature in my opinion. I would keep it shorter, something like this:

```
// Using DefaultError
try {
    val person = io { sampleApiService.fetchPerson() }
} catch (exception: HttpException) {
    val jsonXHttpException = exception.asJsonXHttpException()
    val errors = jsonXHttpException.errors // List<DefaultError>
    // handle errors
    ...
}
```